crates/solver/src/settlement_submission/submitter.rs

```diff
@@ -201,8 +213,22 @@ impl<'a> Submitter<'a> {
             name
         );
 
+        // Remove old submitted transactions with nonce < current nonce
+        self.submitted_transactions.remove_if(name, nonce);
+
+        // Take pending transactions from previous submission loops with the same nonce
+        // Those exist if
+        // 1. Previous loop timed out and no transaction was mined
+        // 2. Previous loop ended with simulation revert, and cancellation tx was sent but not mined
+        let mut transactions = vec![];
+        if let Some(pending_transactions) =
+            self.submitted_transactions
+                .get(name, self.account.address(), nonce)
+        {
+            transactions.extend(pending_transactions.iter());
```

Comment:
This can be `pending_transactions.into_iter()` to avoid copies.
Or maybe a little more succinct:
```rust
        let mut transactions = self
            .submitted_transactions
            .get(name, self.account.address(), nonce)
            .unwrap_or_default();
```